src/core/store_common.jl

```diff
@@ -0,0 +1,210 @@
+# Keep these in sync with the Symbols in src/core/definitions.
+get_store_container_type(::AuxVarKey) = STORE_CONTAINER_AUX_VARIABLES
+get_store_container_type(::ConstraintKey) = STORE_CONTAINER_DUALS
+get_store_container_type(::ExpressionKey) = STORE_CONTAINER_EXPRESSIONS
+get_store_container_type(::ParameterKey) = STORE_CONTAINER_PARAMETERS
+get_store_container_type(::VariableKey) = STORE_CONTAINER_VARIABLES
+
+# Aliases used for clarity in the method dispatches so it is possible to know if writing to
+# DecisionModel data or EmulationModel data
+const DECISION_MODEL_INDEX = Dates.DateTime
+const EMULATION_MODEL_INDEX = Int
```

Comment:
Since they are types I would use `DecisionModelIndexType` and `EmulationModelIndexType`. Or maybe exclude Type. PascalCase, though.